crypto/fipsmodule/bn/rsaz_exp_x2.c

```diff
@@ -62,48 +45,27 @@ OPENSSL_INLINE int number_of_digits(int bitsize, int digit_size)
- *  _ifma256 - uses 256-bit wide IFMA ISA (AVX512_IFMA256)
- */
-
-void ossl_rsaz_amm52x20_x1_ifma256(BN_ULONG *res, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   BN_ULONG k0);
-void ossl_rsaz_amm52x20_x2_ifma256(BN_ULONG *out, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   const BN_ULONG k0[2]);
-void ossl_extract_multiplier_2x20_win5(BN_ULONG *red_Y,
-                                       const BN_ULONG *red_table,
-                                       int red_table_idx1, int red_table_idx2);
-
-void ossl_rsaz_amm52x30_x1_ifma256(BN_ULONG *res, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   BN_ULONG k0);
-void ossl_rsaz_amm52x30_x2_ifma256(BN_ULONG *out, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   const BN_ULONG k0[2]);
-void ossl_extract_multiplier_2x30_win5(BN_ULONG *red_Y,
-                                       const BN_ULONG *red_table,
-                                       int red_table_idx1, int red_table_idx2);
-
-void ossl_rsaz_amm52x40_x1_ifma256(BN_ULONG *res, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   BN_ULONG k0);
-void ossl_rsaz_amm52x40_x2_ifma256(BN_ULONG *out, const BN_ULONG *a,
-                                   const BN_ULONG *b, const BN_ULONG *m,
-                                   const BN_ULONG k0[2]);
-void ossl_extract_multiplier_2x40_win5(BN_ULONG *red_Y,
-                                       const BN_ULONG *red_table,
-                                       int red_table_idx1, int red_table_idx2);
 
+// Dual {1024,1536,2048}-bit w-ary modular exponentiation using prime moduli of
+// the same bit size using Almost Montgomery Multiplication, optimized with
+// AVX512_IFMA256 ISA.
+//
+// The parameter w (window size) = 5.
+//
+//  [out] res      - result of modular exponentiation: 2x{20,30,40} qword
+//                   values in 2^52 radix.
+//  [in]  base     - base (2x{20,30,40} qword values in 2^52 radix)
+//  [in]  exp      - array of 2 pointers to {16,24,32} qword values in 2^64 radix.
+//                   Exponent is not converted to redundant representation.
+//  [in]  m        - moduli (2x{20,30,40} qword values in 2^52 radix)
+//  [in]  rr       - Montgomery parameter for 2 moduli:
+//                     RR(1024) = 2^2080 mod m.
+//                     RR(1536) = 2^3120 mod m.
+//                     RR(2048) = 2^4160 mod m.
+//                   (2x{20,30,40} qword values in 2^52 radix)
+//  [in]  k0       - Montgomery parameter for 2 moduli: k0 = -1/m mod 2^64
+//
+// \return (void).
 static int RSAZ_mod_exp_x2_ifma256(BN_ULONG *res, const BN_ULONG *base,
```

Comment:
That's a good question :) 

At least for `exp`, I'd imagine it's because the function itself operates on its elements while the other parameters are just passed to the assembly implementations. As for while `k0`, I'm not sure there's a good argument beyond "because `rsaz_amm52xYY_x2_ifma256` is defined that way in the header file", which isn't a good enough reason, probably.